src/components/ReviewVoting.jsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from "react";
+import { voteOnReview } from "../api";
+import "../styles/reviewVoting.css";
+
+const ReviewVoting = ({ reviewId, initialVotes }) => {
+  const [votes, setVotes] = useState(initialVotes);
+  const [error, setError] = useState(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleVote = (increment) => {
+    setIsSubmitting(true);
+    setVotes((prevVotes) => increment ? prevVotes + 1 : prevVotes - 1);
+    setError(null);
+
+    voteOnReview(reviewId, increment)
+      .then((review) => {
+        setVotes(review.votes);
+        setIsSubmitting(false);
+      })
+      .catch((error) => {
```

Comment:
a catch block won't work here as the errors have already been caught in the back end. The errors will instead be handled and sent back as a normal response. The errors will be in place of the data, possibly on a key of message (depending on what you did in the back end). To handle this we need to check the response to see if it is data (which means success) or an error message, we can then handle it appropriately.